### PR TITLE
refactor: my collection nav

### PR DESCRIPTION
**Pull request by ds300**
This PR resolves [CX-787]

### Description

This is the path we followed:

- [x] reimplement a skeleton version of the nav hierarchy.
- [x] port existing UI code to work within the new nav hierarchy (preserving the existing components so we can reference them easily while working on this)
- [x] delete the old versions of the components + old nav infra
- [x] QA

cc @damassi 

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-787]: https://artsyproduct.atlassian.net/browse/CX-787

### CODE REVIEW

File: src/lib/Scenes/MyCollection/Screens/MyCollectionArtworkFormModal.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            screenOptions={{
+              headerShown: false,
+              safeAreaInsets: { top: 0, bottom: 0, left: 0, right: 0 },
+              cardStyle: { backgroundColor: "white" },
+            }}
+          >
+            <Stack.Screen name="ArtworkForm" component={MyCollectionArtworkForm} initialParams={{ mode, onDismiss }} />
+            <Stack.Screen name="ArtworkDetailsForm" component={MyCollectionAdditionalDetailsForm} />
+            <Stack.Screen name="AddPhotos" component={MyCollectionAddPhotos} />
+          </Stack.Navigator>
```
Comment (damassi) [inline src/lib/Scenes/MyCollection/Screens/MyCollectionArtworkFormModal.tsx:48]: 👍  👍

File: src/lib/Scenes/MyCollection2/Components/ArtworkFormModal/MyCollectionArtworkFormModal.tsx [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        await myCollectionDeleteArtwork("not an artwork id")
+      } catch (e) {
+        if (__DEV__) {
+          console.error(e)
+        } else {
+          captureException(e)
+        }
+        Alert.alert("An error ocurred", typeof e === "string" ? e : undefined)
+      }
+    })
```
Comment (damassi) [inline src/lib/Scenes/MyCollection2/Components/ArtworkFormModal/MyCollectionArtworkFormModal.tsx:104]: (blocking) @ds300 why are we movingg stuff out of easy peasy? I think this is a separate PR to discuss and definitely not something to mix into a navigation PR.

File: src/lib/Scenes/MyCollection2/Components/ArtworkFormModal/MyCollectionArtworkFormModal.tsx [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    async (path): Promise<string> => {
+      const s3 = await uploadFileToS3(path, acl, assetCredentials)
+      const url = `https://${bucket}.s3.amazonaws.com/${s3.key}`
+      return url
+    }
+  )
+
+  const externalImageUrls: string[] = await Promise.all(uploadPromises)
+  return externalImageUrls
+}
```
Comment (damassi) [inline src/lib/Scenes/MyCollection2/Components/ArtworkFormModal/MyCollectionArtworkFormModal.tsx:168]: Again, all of the concerns were very effectively separated in the model. Now they're mixed into a component. Hard pause on this while we discuss

**Comment by ds300** [comment]
@damassi thanks for raising this. Alas I've found it difficult to refactor the navigation setup to follow normal eigen practices without also refactoring the async code to be more direct. The two are fairly coupled. Once things are working in the new nav hierarchy it should be easier to move these async bits back into easy-peasy and we can revisit then. I appreciate how thoughtful and well-organised the My Collections easy-peasy code is, makes our lives a lot easier 🙏

**Comment by damassi** [comment]
Ok that makes sense. Sorry for being slightly alarmed here! Carry on.

File: src/lib/store/migration.ts [hunk 4]
```diff
@@ -31,6 +32,9 @@ export const artsyAppMigrations: Migrations = {
     state.myCollection = state.consignments
     delete state.consignments
   },
+  [Versions.RemoveMyCollectionNavigationState]: (state) => {
+    delete state.myCollection.navigation
+  },
```
Comment (brainbicycle) [inline src/lib/store/migration.ts:37]: 🎉

File: src/lib/Scenes/MyCollection/State/MyCollectionArtworkModel.tsx [hunk 5]
```diff
@@ -90,19 +87,6 @@ export const MyCollectionArtworkModel: MyCollectionArtworkModel = {
...
     artworkErrorOccurred: false,
-
-    /**
-     * TODO: this will likely be able to go away once we update our mutations to take
-     * advantage of the new Relay v10 directive-based update model.
-     * See https://github.com/facebook/relay/releases/tag/v10.0.0.
-     *
-     * The relay global ID of the `me` field, used to insert / delete edge post mutation.
-     */
-    meGlobalId: "",
```
Comment (damassi) [inline src/lib/Scenes/MyCollection/State/MyCollectionArtworkModel.tsx:135]: How does this work now without the relay id to update the node?

**Comment by damassi** [comment]
Amazing git discipline throughout this ya'll 👍 Its been very easy to follow changes commit by commit. (And nice work on the refactor.)

File: src/lib/Scenes/MyCollection/State/MyCollectionArtworkModel.tsx (hunk 5 above)
Comment (ds300) [reply src/lib/Scenes/MyCollection/State/MyCollectionArtworkModel.tsx:135]: We no longer update the relay store imperatively. Instead we use the normal cache updating behaviour for artwork updates, and we refetch the artwork list after any updates, to handle adding/removing.

Comment (damassi) [reply src/lib/Scenes/MyCollection/State/MyCollectionArtworkModel.tsx:135]: Makes sense 👍

File: src/lib/utils/AboveTheFoldQueryRenderer.tsx [hunk 6]
```diff
@@ -83,7 +83,9 @@ export function AboveTheFoldQueryRenderer<AboveQuery extends OperationType, Belo
         query={props.above.query}
         variables={props.above.variables}
         render={(args) => {
-          setAboveArgs(args)
+          setImmediate(() => {
+            setAboveArgs(args)
+          })
```
Comment (ds300) [inline src/lib/utils/AboveTheFoldQueryRenderer.tsx:88]: the changes in this file fix a warning that I think was probably introduced in the recent react-native upgrade

File: src/lib/Scenes/MyCollection/Screens/ArtworkFormModal/Screens/MyCollectionArtworkFormAddPhotos.tsx [hunk 7]
```diff
@@ -1,56 +1,62 @@
...
+const MARGIN = 20
+
+export const MyCollectionAddPhotos: React.FC<StackScreenProps<ArtworkFormModalScreen, "AddPhotos">> = ({
+  navigation,
+}) => {
   const formValues = AppStore.useAppState((state) => state.myCollection.artwork.sessionState.formValues)
-  const imageSize = useImageSize()
   const { photos } = formValues
+  const { width: screenWidth } = useScreenDimensions()
+  const numColumns = isPad() ? 5 : 2
```
Comment (MounirDhahri) [inline src/lib/Scenes/MyCollection/Screens/ArtworkFormModal/Screens/MyCollectionArtworkFormAddPhotos.tsx:22]: looks good on iPad as well 🙇‍♂️

File: src/lib/store/migration.ts (hunk 4 above)
Comment (MounirDhahri) [reply src/lib/store/migration.ts:37]: 🎉

**Review by MounirDhahri — APPROVED** [review]

**Comment by MounirDhahri** [comment]
Really good work here  🔥🔥🔥

**Pull request merged**